orchestration/dagster_orchestration/hca_orchestration/contrib/argo_workflows.py

```diff
@@ -0,0 +1,67 @@
+from argo.workflows.client import ApiClient as ArgoApiClient,\
+                                  ArchivedWorkflowServiceApi,\
+                                  Configuration as ArgoConfiguration
+from argo.workflows.client.models import V1alpha1Workflow
+
+
+def generate_argo_archived_workflows_client(host_url, access_token):
+    return ArchivedWorkflowServiceApi(
+        api_client=ArgoApiClient(
+            configuration=ArgoConfiguration(host=host_url),
+            header_name="Authorization",
+            header_value=f"Bearer {access_token}"))
+
+
+class ArgoArchivedWorkflowsClientMixin:
+    def __init__(self, *args, argo_url, access_token, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.argo_url = argo_url
+        self.access_token = access_token
+        self._client = None
+
+    def client(self):
+        if not self._client:
+            self._client = generate_argo_archived_workflows_client(self.argo_url, self.access_token)
+
+        return self._client
+
+    def list_archived_workflows(self):
+        self._pull_paginated_results(self.client().list_archived_workflows)
```

Comment:
How performant is this endpoint? ie.,. can get we get all the archived workflows in the sensor polling interval or is there some danger of the sensor lapping itself?

Reply:
Also, what happens when a sensor laps itself? Does dagster drop the evaluation on the floor or will the sensor evaluations queue up?

Reply:
argo returned a page of results in about a quarter second when i was testing, so i don't expect delays to impact us compared to something like redcap.

as for a sensor lapping itself, it looks like it'll spin up a container to run it regardless of whether the previous run is complete yet, so it's possible to have multiple simultaneous runs for the same sensor. the design appears to be set up to ensure that no executions get dropped. dagster's by-run-name deduplication avoids this creating duplicate runs, but it's a concern for potentially exhausting resources if things start running unusually long for some reason.

Reply:
checked in with the dagster team in their slack, it turns out the behavior is actually that the sensor interval doesn't start counting until the current evaluation of the sensor completes, so if it takes ten minutes to evaluate, it'll wait thirty seconds after that ten minute run completes before starting another.